patches/minecraft/net/minecraft/entity/item/BoatEntity.java.patch

```diff
@@ -30,3 +69,62 @@
     public static enum Status {
        IN_WATER,
        UNDER_WATER,
+@@ -786,22 +782,51 @@
+       IN_AIR;
+    }
+ 
+-   public static enum Type {
+-      OAK(Blocks.field_196662_n, "oak"),
+-      SPRUCE(Blocks.field_196664_o, "spruce"),
+-      BIRCH(Blocks.field_196666_p, "birch"),
+-      JUNGLE(Blocks.field_196668_q, "jungle"),
+-      ACACIA(Blocks.field_196670_r, "acacia"),
+-      DARK_OAK(Blocks.field_196672_s, "dark_oak");
++   public static enum Type implements net.minecraftforge.common.IExtensibleEnum {
++      OAK(Blocks.field_196662_n, "oak") { public Item getBoatItem() { return Items.field_151124_az; } },
++      SPRUCE(Blocks.field_196664_o, "spruce") { public Item getBoatItem() { return Items.field_185150_aH; } },
++      BIRCH(Blocks.field_196666_p, "birch") { public Item getBoatItem() { return Items.field_185151_aI; } },
++      JUNGLE(Blocks.field_196668_q, "jungle") { public Item getBoatItem() { return Items.field_185152_aJ; } },
++      ACACIA(Blocks.field_196670_r, "acacia") { public Item getBoatItem() { return Items.field_185153_aK; } },
++      DARK_OAK(Blocks.field_196672_s, "dark_oak") { public Item getBoatItem() { return Items.field_185154_aL; } };
+ 
+       private final String field_184990_g;
+       private final Block field_195934_h;
++      private final net.minecraft.util.IItemProvider[] drops;
++      private Item boatItem;
+ 
+       private Type(Block p_i48146_3_, String p_i48146_4_) {
+          this.field_184990_g = p_i48146_4_;
+          this.field_195934_h = p_i48146_3_;
++         this.drops = new net.minecraft.util.IItemProvider[]{ Items.field_151055_y, Items.field_151055_y, p_i48146_3_, p_i48146_3_, p_i48146_3_ };
+       }
+ 
++      private Type(Block planks, String name, net.minecraft.util.IItemProvider[] drops) {
++         this.field_184990_g = name;
++         this.field_195934_h = planks;
++         this.drops = drops;
++      }
++
++      public static Type create(String enumName, Block planks, String name) {
++         throw new IllegalStateException("Enum not extended");
++      }
++
++      public static Type create(String enumName, Block planks, String name, net.minecraft.util.IItemProvider[] drops) {
++         throw new IllegalStateException("Enum not extended");
++      }
++
++      public void setBoatItem(Item boatItem) {
++         this.boatItem = boatItem;
++      }
++
++      public Item getBoatItem() {
++         return (this.boatItem == null) ? OAK.boatItem : this.boatItem;
++      }
++
++      public net.minecraft.util.IItemProvider[] getDrops() {
++         return this.drops;
```

Comment:
This should return an immutable view or keep the items as an immutable collection. Also, might be more flexible to use itemstacks?

Reply:
Immutable set and the usage of itemstacks are implemented now. Thanks for all your feedback!